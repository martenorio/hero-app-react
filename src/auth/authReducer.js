import { types } from "../types/types";

export const authReducer = (state = {}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logout:
            return {
                logged: false
            }
        default:
            return state;
    }
}

// example
// const state = {
//     name:'Miguel',
//     logged:true
// }
// example
// const loginAction = {
//     type: types.login,
//     payload: {
//         name:'miguel',
//         email: 'miguel@gmail.com'
//     }
// }