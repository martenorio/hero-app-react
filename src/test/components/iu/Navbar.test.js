import { mount } from "enzyme";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();

//El mock es necesario, por que dentro de la función 
//se ejecuta navigate de  useNavigate que forma parte de 
//react-router-dom y para que la prueba 
//no tenga errores con la funcion de navigate se simula la ejecución
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}))

describe('Navbar test', () => {
    const contextValue = {
        user:{
            logged: true,
            name: 'John'
        }, 
        dispatch : jest.fn()
    }
    const wrapper = mount( 
        <AuthContext.Provider value={contextValue} >
            <MemoryRouter initialEntries={ ['/'] }>
                <Routes>
                    <Route path="/" element={<Navbar />}></Route>
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    );
    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toEqual('John');
    })
    test('should call logout, navigate, and dispatch with arguments',()=>{
        // wrapper.find('button').prop('onClick')();
        wrapper.find('button').simulate('click');
        expect(contextValue.dispatch).toHaveBeenCalled();
        expect(contextValue.dispatch).toHaveBeenCalledWith({'type':types.logout});
        expect(mockNavigate).toHaveBeenCalled();
        expect(mockNavigate).toHaveBeenCalledWith('/login',{ replace: true });
    })
})