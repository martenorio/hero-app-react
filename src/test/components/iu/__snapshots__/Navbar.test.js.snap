// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test should display correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link false"
        exact="true"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav-item nav-link false"
        exact="true"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav-item nav-link false"
        exact="true"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
         
        John
         
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
